File: src/com/leetcode/Easy/MinimumDepthOfBinaryTree.java
package com.leetcode.Easy;

// 111. Minimum Depth of Binary Tree
public class MinimumDepthOfBinaryTree {
    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode(int x) {
            val = x;
        }
    }

    public int minDepth(TreeNode root) {
        if(root == null)
            return 0;

        int leftDepth = minDepth(root.left);
        int rightDepth = minDepth(root.right);
        if(leftDepth == 0) {
            return 1 + rightDepth;
        } else if(rightDepth == 0) {
            return 1 + leftDepth;
        }

        return 1 + Math.min(leftDepth, rightDepth);
    }
}

File: JavaScript/Easy/MinimumDepthOfBinaryTree.js
/**
 * 111. Minimum Depth of Binary Tree
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
const minDepth = function (root) {
  if (root == null) return 0;

  let leftDepth = minDepth(root.left);
  let rightDepth = minDepth(root.right);
  if (leftDepth == 0) {
    return 1 + rightDepth;
  } else if (rightDepth == 0) {
    return 1 + leftDepth;
  }

  return 1 + Math.min(leftDepth, rightDepth);
};
